components/tidb_query_datatype/src/codec/collation/collator/latin1_bin.rs

```diff
@@ -0,0 +1,35 @@
+// Copyright 2020 TiKV Project Authors. Licensed under Apache-2.0.
+
+use super::*;
+use bstr::{ByteSlice, B};
+
+/// Collator for latin1_bin collation with padding behavior (trims right spaces).
+#[derive(Debug)]
+pub struct CollatorLatin1Bin;
+
+impl Collator for CollatorLatin1Bin {
+    type Charset = CharsetBinary;
+
+    #[inline]
+    fn validate(_bstr: &[u8]) -> Result<()> {
+        Ok(())
+    }
+
+    #[inline]
+    fn write_sort_key<W: BufferWriter>(writer: &mut W, bstr: &[u8]) -> Result<usize> {
+        let s = B(bstr).trim_end();
```

Comment:
I think we should not trim tabs, please check the behavior in tests.

Reply:
Good catch, fixed.